ckanext/recombinant/views.py

```diff
@@ -374,25 +370,26 @@ def _schema_json(dataset_type, published_resource=False):
             fld = OrderedDict()
             resource['fields'].append(fld)
             fld['id'] = field['datastore_id']
-            if fld['id'] in pkeys:
-                fld['obligation'] = 'mandatory'
-            elif field.get('excel_required'):
-                fld['obligation'] = 'mandatory'
-            elif field.get('excel_required_formula'):
-                fld['obligation'] = 'conditional'
-            else:
-                fld['obligation'] = 'optional'
+            fld['obligation'] = OrderedDict()
```

Comment:
OrderedDict was an old python thing. In python3.6+ regular dicts are ordered by insertion.

Reply:
changed all `OrderedDict` to `{}`